Assets/Scripts/Interceptors/CarrierInterceptor.cs

```diff
@@ -34,27 +34,25 @@ protected override void DrawDebugVectors() {
 
   private void SpawnSubmunitions() {
     List<Interceptor> submunitions = new List<Interceptor>();
-    for (int i = 0; i < dynamicAgentConfig.submunitions_config.num_submunitions; ++i) {
-      DynamicAgentConfig convertedConfig = DynamicAgentConfig.FromSubmunitionDynamicAgentConfig(
-          dynamicAgentConfig.submunitions_config.dynamic_agent_config);
-      InitialState initialState = new InitialState();
-      initialState.position = transform.position;
+    for (int i = 0; i < agentConfig.SubmunitionsConfig.NumSubmunitions; ++i) {
+      Configs.AgentConfig submunitionsConfig = agentConfig.SubmunitionsConfig.AgentConfig;
+      Simulation.State initialState = new Simulation.State();
+      initialState.Position = Coordinates3.ToProto(transform.position);
```

Comment:
We now pull straight from `agentConfig.SubmunitionsConfig`, but protobuf leaves this field null when a carrier interceptor omits submunitions. The previous JSON path always materialised an empty object via `ProtobufInitializer`, so it never null-dereferenced. Today the first carrier without submunitions will hit a `NullReferenceException` on the loop guard. 

Consider watching for `null` (and zero-count) before iterating?